esp/services/esdl_svc_engine/esdl_svc_engine.cpp

```diff
@@ -39,6 +39,70 @@ CEsdlSvcEngineSoapBindingEx::CEsdlSvcEngineSoapBindingEx(IPropertyTree* cfg, con
 {
 }
 
+IPropertyTree *createContextMethodConfig(IPropertyTree *methodConfig)
+{
+    const char *include = "*";
+    if (methodConfig->hasProp("@contextInclude"))
+        include = methodConfig->queryProp("@contextInclude");
+    const char *remove = "Transforms|xsdl:CustomRequestTransform";
+    if (methodConfig->hasProp("@contextRemove"))
+        remove = methodConfig->queryProp("@contextRemove");
+    const char *removeAttrs = "@contextInclude|@contextRemove|@contextAttRemove";
+    if (methodConfig->hasProp("@contextAttRemove"))
+        removeAttrs = methodConfig->queryProp("@contextAttRemove");
+
+    Owned<IPropertyTree> contextConfig;
+    if (include && !streq(include, "*"))
+    {
+        contextConfig.setown(createPTree("Method", ipt_ordered));
+        Owned<IAttributeIterator> aiter = methodConfig->getAttributes(); //for now include all attributes
+        ForEach (*aiter)
+            contextConfig->addProp(aiter->queryName(), aiter->queryValue());
+
+        if (*include) //if *include==0 we should have an empty tree
+        {
+            StringArray xpaths;
+            xpaths.appendListUniq(include, "|", true); //don't supported quoted | for now
+            ForEachItemIn(pos, xpaths)
+            {
+                Owned<IPropertyTreeIterator> toInclude = methodConfig->getElements(xpaths.item(pos));
+                ForEach(*toInclude)
+                    contextConfig->addPropTree(toInclude->query().queryName(), LINK(&toInclude->query()));
+            }
+        }
+    }
+    if (remove && *remove)
```

Comment:
I would expect that removing the method element would also remove the method element's attributes. Attributes have no meaning outside the context of their element. It would be incredibly counter intuitive to omit the entire branch but have it included because the attributes were not explicitly marked for removal.

That doesn't mean a contextAttRemove="*" doesn't have value on its own.

Reply:
As we discussed offline, contextRemove="*" affects the children of our root node which in this case is "Method".  So the Method tag and its attributes still exist.  I remove method only if it is completely empty.

But as we also discussed offline I will use contextRemove="." to mean remove the entire current (root) node.  Which will include the attributes.